src/rules/__tests__/no-deprecated-functions.test.ts

```diff
@@ -3,20 +3,36 @@ import * as os from 'os';
 import * as path from 'path';
 import { JSONSchemaForNPMPackageJsonFiles } from '@schemastore/package';
 import { TSESLint } from '@typescript-eslint/experimental-utils';
+import rimraf from 'rimraf';
 import rule, {
   JestVersion,
   _clearCachedJestVersion,
 } from '../no-deprecated-functions';
 
+const root = process.cwd();
+afterEach(() => process.chdir(root));
```

Comment:
`process.chdir` is not actually supported by Jest. It _works_ since it's the real `chdir`, but it breaks a bunch of Jest's assumptions. Since it actually changes dir, it leaks between test files and the jest process itself.

It's not new though, so no need to change now. But something to keep in mind 🙂

Reply:
Yeah, which is what's happening here - turns out the tests I wrote for the rule are what's been causing the weird CI failure.

I'm not sure why it only happens on Node 8 & 10, but the issue is that by changing the cwd to be outside of the projects folder, when looking for plugins `eslint` looks into the `node_modules` in `/tmp/eslint-plugin-jest-<tmpdir>` while using the config from `eslint-plugin-jest`, and so fails to find all its plugins.

I hacked this together to confirm, so now I'll break it out into another PR, and then after that I should be able to merge & ship a bunch of our open PRs.

Reply:
On an aside, is there anything we could do in `jest` core to improve `process.chdir` support?

One such idea that comes to mind would be to do that I'm doing here: "pin" the cwd as part of `jest`'s setup phase, and then somehow make that available in places.

I'm not sure how worth it would be, but if there are any low hanging fruit that could be done to even just reduce a few of the assumptions, I'd be happy to explore :)